parasoft-findings-teamcity-agent/src/main/java/com/parasoft/findings/teamcity/agent/ParasoftFindingsBuildProcess.java

```diff
@@ -156,63 +160,66 @@ private void doWork() {
             _build.getBuildLogger().error("No reports found for pattern: "+reportsLocation);
         } else {
             for (File from : reports) {
-                ReportParserDescriptor rpd = getReportParserDescriptor(from);
-                if (rpd != null) {
+                List<ReportParserDescriptor> rpds = getReportParserDescriptors(from);
+                if(rpds.isEmpty()) {
+                    _build.getBuildLogger().warning("Skipping unrecognized report file: " + from.getAbsolutePath());
+                    continue;
+                }
+                for(ReportParserDescriptor rpd : rpds) {
                     _build.getBuildLogger().message("Transforming "+from.getAbsolutePath()+" with "+rpd.getLabel());
-                    String targetFileName = PREFIX + from.getName();
+                    String targetFileName = rpd.getOutputFileNamePrefix() + from.getName();
                     File to = new File(from.getParentFile(), targetFileName);
                     transform(from, to, rpd.getXSL(), checkoutDir);
-                } else {
-                    _build.getBuildLogger().warning("Skipping unrecognized report file: " + from.getAbsolutePath());
                 }
             }
         }
     }
 
-    private ReportParserDescriptor getReportParserDescriptor(File from) {
-        StreamSource xml = new StreamSource(from);
-        XMLEventReader reader = null;
+    private List<ReportParserDescriptor> getReportParserDescriptors(File from) {
+        List<ReportParserDescriptor> descriptors = new ArrayList<ReportParserDescriptor>();
```

Comment:
If I didn't get it wrong, unit test results and static analysis results could be generated by one tool in one report file, but SOAtest test results will be in separate file. If that's the case, when the report is a SOAtest report, it should not need to check for other descriptors. I think that's the reason why the original logic is only picking one of the descriptor.

Reply:
Even though it is not a common use case, SOAtest reports can contain both functional test results and static analysis results. So it would be nice to still support that if possible, although we don't need to spend much time making sure it works or testing it.

Reply:
I see, in that way we'll go through all descriptors for each file. 
There's a small fix needed on line 195 then: we forgot to add the descriptor to the list.